modules/apps/transfer/keeper/relay_test.go

```diff
@@ -385,22 +404,25 @@ func (suite *KeeperTestSuite) TestOnRecvPacket() {
 			err = suite.chainB.GetSimApp().TransferKeeper.OnRecvPacket(suite.chainB.GetContext(), packet, data)
 
 			// check total amount in escrow of received token denom on receiving chain
-			var (
-				denom       string
-				totalEscrow sdk.Coin
-			)
-			if tc.recvIsSource {
-				denom = sdk.DefaultBondDenom
-			} else {
-				denom = trace.IBCDenom()
-			}
-			totalEscrow = suite.chainB.GetSimApp().TransferKeeper.GetTotalEscrowForDenom(suite.chainB.GetContext(), denom)
+			totalEscrow := suite.chainB.GetSimApp().TransferKeeper.GetTotalEscrowForDenom(suite.chainB.GetContext(), sdk.DefaultBondDenom)
 			suite.Require().Equal(expEscrowAmount, totalEscrow.Amount)
 
 			if tc.expPass {
 				suite.Require().NoError(err)
+
+				if tc.recvIsSource {
+					_, found := suite.chainB.GetSimApp().BankKeeper.GetDenomMetaData(suite.chainB.GetContext(), denomTraceOnB.IBCDenom())
+					suite.Require().False(found)
+				} else {
+					denomMetadata, found := suite.chainB.GetSimApp().BankKeeper.GetDenomMetaData(suite.chainB.GetContext(), denomTraceOnB.IBCDenom())
+					suite.Require().True(found)
+					suite.Require().Equal(expDenomMetadataOnB, denomMetadata)
+				}
 			} else {
 				suite.Require().Error(err)
+
+				_, found := suite.chainB.GetSimApp().BankKeeper.GetDenomMetaData(suite.chainB.GetContext(), denomTraceOnB.IBCDenom())
```

Comment:
There is one test failing for this check. But I need some help to understand why.

Reply:
Checks on errors are unnecessary as state reversals will occur by baseapp. I removed the assertion, I don't think this situation justifies the extra hassle